src/store/modules/user/actions.ts

```diff
@@ -69,6 +70,12 @@ const actions: ActionTree<UserState, RootState> = {
         return uniqueFacilities
       }, []);
 
+      const facilityId = router.currentRoute.value.query.facilityId
+      if (facilityId) {
+        const facility = userProfile.facilities.find((facility: any) => facility.facilityId === facilityId);
```

Comment:
Add check that if the facility passed in the routes is not found then do continue the apps login flow by default and add a toast message that `redirecting to home page due to incorrect information being passed.` (something like this).